Docs/ROADMAP.md

```diff
@@ -0,0 +1,328 @@
+
+## Manafest: Arena 2
+
+The sequel shares the premise of the previous game. The player selects a character with unique abilities. Given a tree of encounters, they choose their own path up the tree and advance to the end of the game by completing encounters.
+
+### Menus and flow overview
+
+The following is a rundown of features and menus the player will interact with.
+
+#### Main menu
+
+The player is presented with the game's logo, background music, and buttons to navigate to other menus.
+- Continue: Present only if a game has been started
+- New: Go to new game menu
+- Settings: Go to settings menu
+- Credits: Go to credits menu
+- Quit: Exit the game
+
+#### Settings Menu
+
+This menu provides some settings
+
+- Master Volume: Volume for all sound 
+- Sound Effects Volume: Volume for sound effects
+- Music Volume: Volume for in-game music
+- Mouse sensitivity X: Sensitivity for turning left and right
+- Mouse Sensitivity y: Sensitivity for looking up and down
+- Name: Player's name
+
+There are also buttons on this menu:
+Main Menu: Return to main menu
+Revert: reload saved settings
+Save: Save current settings
+Controls: Navigate to controls menu
+
+#### Controls menu
+
+This menu provides various actions and the inputs they are mapped to.
+The user can click an action, then a new key to edit the mapping.
+
+- There are then some buttons on this menu:
+- Back: returns to options menu
+- Revert: load saved settings
+- Save: save current settings
+
+#### Credits menu
+
+A list of credits identifies all contributors and any special thanks.
+
+There is additionally a single button to return to the main menu.
+
+#### New game menu
+
+This menu shows the selection of the three starting characters. The player should be able to select each character to see information about them.
+
+There are buttons on this page:
+- Character selection buttons: When pressed highlight character and display information.
+- Main menu: Return to main menu
+- Start game: Appears when a character is selected. Creates a new save and navigates to the Career map.
+
+#### Career Menu
+
+This menu displays the player's progression through their career in the arena as an inverted tree of encounters.
+The tree's root node is at the top level of the menu, and its leaves are at the bottom of the menu.
+Each encounter is represented by a button that is disabled until active.
+To start with, only the leaf nodes of this tree on the bottom level are active. After playing through one encounter
+on a level, the parent nodes of that encounter become active. This is functionally identical to the map in Slay The Spire.
+The entire tree is randomly generated with different nodes upon the start of a new game.
+
+If the map has more levels than will fit on the screen, the player can scroll vertically by moving their mouse to the top or
+bottom center of the screen.
+
+There is also a button to return to the main menu.
```

Comment:
Most of these menus are straighforward, but it might be nice to have a wire frame (not even styled perhaps) to accompany the more complex menus (this one specifically I think I get the gist of it).
You can even use relative file linking in the `.md` file to link it all nicely together! (or embed it as an image :3)

Reply:
I'll have to look into that. I've never made wireframes before.

Reply:
https://www.lucidchart.com was really nice, used it in college and at my last job. Really full features, quick (though it's web only/js I believe), and doesn't fall apart when you get to really complex stuff.

http://dia-installer.de/ was a GPL licensed one I've used before lucidchart. I liked it just about as much as lucidchart but found it broke down in certain complex grouping scenarios.